pkg/sql/conn_executor_test.go

```diff
@@ -1230,6 +1230,82 @@ CREATE TABLE t1.test (k INT PRIMARY KEY, v TEXT);
 	})
 }
 
+// TestShowTransferStateError tests that the SHOW TRANSFER STATE statement
+// will return an error as a SQL value whenever a tenant server is not used.
+// For actual tests, see pkg/ccl/testccl/sqlccl.
+func TestShowTransferStateError(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	params := base.TestServerArgs{}
+	s, sqlConn, _ := serverutils.StartServer(t, params)
+	defer s.Stopper().Stop(ctx)
+
+	_, err := sqlConn.Exec("SELECT 1")
+	require.NoError(t, err)
+
+	t.Run("without_transfer_key", func(t *testing.T) {
+		rows, err := sqlConn.Query("SHOW TRANSFER STATE")
+		require.NoError(t, err, "show transfer state failed")
+		defer rows.Close()
+
+		resultColumns, err := rows.Columns()
+		require.NoError(t, err)
+
+		const expectedNumColumns = 3
+		if len(resultColumns) != expectedNumColumns {
```

Comment:
nit: assert the content of the columns instead of the number of columns.
```
require.Equal(t, ["error", "session_state_base64", "session_revival_token_base64"], resultColumns)
```